win32ss/user/user32/controls/listbox.c

```diff
@@ -321,9 +321,14 @@ static LRESULT LISTBOX_SetTopItem( LB_DESCR *descr, INT index, BOOL scroll )
         }
         else
             diff = (descr->top_item - index) * descr->item_height;
-
-        ScrollWindowEx( descr->self, 0, diff, NULL, NULL, 0, NULL,
-                        SW_INVALIDATE | SW_ERASE | SW_SCROLLCHILDREN );
+#ifdef __REACTOS__
+        if (descr->style & LBS_MULTICOLUMN)
+            ScrollWindowEx(descr->self, diff, 0, NULL, NULL, 0, NULL,
+                           SW_INVALIDATE | SW_ERASE | SW_SCROLLCHILDREN);
+        else
+#endif
```

Comment:
If the `descr->style & LBS_MULTICOLUMN` condition is not met, you will literally jump to the other block with no code hence it doesn't know which condition to check on.

You probably wanted to mean `#else`, the condition directive.

Reply:
The  else just exist when defined __REACTOS__, and in that case code jumps to the old instruction 
ScrollWindowEx(...) . It works fine for me, without erros.
No need #else directive

Reply:
You seem to misunderstand what the preprocessor directives are for.

When you wrap a portion of code within a certain directive define, i.e. `__REACTOS__`, the compiler will only compile the portion of the said code depending on the compiler build target (or whatever it's called). The compiler will literally compile this:
```C
if (descr->style & LBS_MULTICOLUMN)
            ScrollWindowEx(descr->self, diff, 0, NULL, NULL, 0, NULL,
                           SW_INVALIDATE | SW_ERASE | SW_SCROLLCHILDREN);
else
```
Having an empty `else {...}` block is not illegal but it's ugly and hard to maintain as other contributors might not have a clue the intention behind the empty block. Just because the code doesn't give compile errors and the like doesn't mean the code in question is always correct. So, I repeat myself again, remove that `else` and use the `#else` directive.

Reply:
After the preprocessor shouldn't it be like this?
```
if (descr->style & LBS_MULTICOLUMN)
            ScrollWindowEx(descr->self, diff, 0, NULL, NULL, 0, NULL,
                           SW_INVALIDATE | SW_ERASE | SW_SCROLLCHILDREN);
else
            ScrollWindowEx(descr->self, 0, diff, NULL, NULL, 0, NULL,
                           SW_INVALIDATE | SW_ERASE | SW_SCROLLCHILDREN);
```

Reply:
You say:
"I repeat myself again, remove that else and use the #else directive."

@fraizeraust, with the utmost respect, I ask you to observe.
^#ifdefined __REACTS__
     if (describe-> style & LBS_MULTICOLUMN)
            ScrollWindowEx (describe-> self, diff, 0, NULL, NULL, 0, NULL,
                           SW_INVALIDATE | SW_ERASE | SW_SCROLLCHILDREN);
#else
            ScrollWindowEx (describe-> self, 0, diff, NULL, NULL, 0, NULL,
                           SW_INVALIDATE | SW_ERASE | SW_SCROLLCHILDREN);
#endif
^
This portion of code does not achieve the required objectives, I cannot remove the else statement. The third form would be:
^
#ifdef __REACTS__
        if (describe-> style & LBS_MULTICOLUMN)
            ScrollWindowEx (describe-> self, diff, 0, NULL, NULL, 0, NULL,
                           SW_INVALIDATE | SW_ERASE | SW_SCROLLCHILDREN);
        else
            ScrollWindowEx (describe-> self, 0, diff, NULL, NULL, 0, NULL,
                           SW_INVALIDATE | SW_ERASE | SW_SCROLLCHILD
#else
             ScrollWindowEx (describe-> self, 0, diff, NULL, NULL, 0, NULL,
                           SW_INVALIDATE | SW_ERASE | SW_SCROLLCHILDREN);
#endif
^
Without a doubt, I find the first form much more elegant.
To break this discussion I am going to ask the opinion of the person who requested this modification Mr. @ Doug-Lions.

Reply:
Tu dizes:
" I repeat myself again, remove that else and use the #else directive."

@fraizeraust, com o maior respeito, peço que observe.
#ifdef __REACTOS__
`if (descr->style & LBS_MULTICOLUMN)
            ScrollWindowEx(descr->self, diff, 0, NULL, NULL, 0, NULL,
                           SW_INVALIDATE | SW_ERASE | SW_SCROLLCHILDREN);
`
#else 
`if (descr->style & LBS_MULTICOLUMN)
            ScrollWindowEx(descr->self, diff, 0, NULL, NULL, 0, NULL,
                           SW_INVALIDATE | SW_ERASE | SW_SCROLLCHILDREN);
`
#endif 
esta porção de código não atinge os objectivos necessários, não posso remover a instrução else. A terceira forma seria:

Reply:
Diverging opinions on code elegance. To break the tie I will ask the opinion of who requested this modification. Mr. @Doug-Lions what do you think about this discussion?

Reply:
Yo guys.
The original code (before @zecarlos1957) was literally:
```c
ScrollWindowEx( descr->self, 0, diff, NULL, NULL, 0, NULL,
                SW_INVALIDATE | SW_ERASE | SW_SCROLLCHILDREN );
```
Now, and only because this code will be compiled using our build system where the `__REACTOS__` preprocessor symbol is defined, what we will compile after the @zecarlos1957 modifications is this:
```c
        if (descr->style & LBS_MULTICOLUMN)
            ScrollWindowEx(descr->self, diff, 0, NULL, NULL, 0, NULL,
                           SW_INVALIDATE | SW_ERASE | SW_SCROLLCHILDREN);
        else
            ScrollWindowEx(descr->self, 0, diff, NULL, NULL, 0, NULL,
                           SW_INVALIDATE | SW_ERASE | SW_SCROLLCHILDREN);
```
and that's OK by itself, if this is the wanted behaviour, namely:
- if the list has the multicolumn style, scroll in the 'x' direction (horizontal) only with an amount == `diff`,
- otherwise (if the list does NOT have the multicolumn style), use the old behaviour of scrolling in the 'y' (vertical) direction with the amount == `diff`.

Reply:
@zecarlos1957 you proposed code will work fine as noted by @HBelusca above. It may be easier to read as shown below, but what you have will work as well. Thanks.
`#ifdef __REACTOS__
        if (descr->style & LBS_MULTICOLUMN)
            ScrollWindowEx(descr->self, diff, 0, NULL, NULL, 0, NULL,
                           SW_INVALIDATE | SW_ERASE | SW_SCROLLCHILDREN);
        else
            ScrollWindowEx(descr->self, 0, diff, NULL, NULL, 0, NULL,
                           SW_INVALIDATE | SW_ERASE | SW_SCROLLCHILDREN);
#else
        ScrollWindowEx( descr->self, 0, diff, NULL, NULL, 0, NULL,
                        SW_INVALIDATE | SW_ERASE | SW_SCROLLCHILDREN );
#endif`

Reply:
Thank you for your participation